### PR TITLE
Some examples of Feature API migrations

### PR DESCRIPTION
**Pull request by thecoop**

### CODE REVIEW

**Comment by thecoop** [comment]
@dnhatn This changes the semantics for ESQL stats action from 'only on new nodes' to 'only work when all nodes are upgraded'. This follows the ES compatibility guarantees that new features are only valid when all nodes are upgraded.

**Review by dnhatn — COMMENTED** [review]
@thecoop I left two comments regarding the ESQL stats feature. Thank you for working on this **feature**.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import org.elasticsearch.Version;
+import org.elasticsearch.features.FeatureSpecification;
+import org.elasticsearch.features.NodeFeature;
+
+import java.util.Map;
+
+public class EsqlFeatures implements FeatureSpecification {
+    @Override
+    public Map<NodeFeature, Version> getHistoricalFeatures() {
+        return Map.of(TransportEsqlStatsAction.ESQL_STATS_FEATURE, Version.V_8_10_0);
```
[the remark this answers is not in the document]
Comment (thecoop) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java:19]: Oh yes, oops

**Comment by thecoop** [comment]
Moved to separate PRs #101530 #101531

**Pull request closed without merging**